hazelcast/src/main/java/com/hazelcast/map/impl/operation/SetOperation.java

```diff
@@ -42,7 +42,7 @@ public void afterRun() {
 
     @Override
     public void run() {
-        newRecord = recordStore.set(dataKey, dataValue, ttl);
```

Comment:
Perhaps one way to reduce the complexity is to add a telescoping method where the 'countAsAccess' is true. So that code that doesn't care about it like the putAll/set etc don't need to be modified.

Reply:
Seems like that is the way to go. The only problem is the name of this method, I will try to invent a good one.

Reply:
Fixed.